### DISCUSSION TITLE
Output header files .head

### DISCUSSION THREAD

**Discussion by tosaric** [astromatic/scamp#26] [Q&A]
How to read .head file? Fits complains that it is not valid fits file, and I want to read it so I can create wcs and use it to convert pixel coordinates to sky coordinates

**Comment by hbouy** [comment]
it’s a regular ascii file, not a FITS file.

**Comment by ebertin** [comment]
It is indeed an ASCII file, each line ending with a linefeed (e.g.,on Linux).

Note that both SWarp and SExtractor override the values of the FITS keywords of an input FITS image header with those from a .head file with the same prefix located in the same directory (you can check that such a .head file has been taken into account by looking for the `EXT. HEADER` tag in the info output from both packages). Therefore if you want to get the calibrated world coordinates in a source catalog, all you have to do is run SExtractor on the original image, while making sure that the .head file from SCAMP is present in the same directory.

Now, you may also want to transfer the WCS parameters from the .head file to the original image file. For this you may either use the [missfits](https://astromatic.net/software/missfits) package, or if you prefer to use Python, read the .head file with the [Header.fromfile](https://docs.astropy.org/en/stable/io/fits/api/headers.html#astropy.io.fits.Header.fromfile) method from the io.fits module of AstroPy.

**Comment by tosaric** [comment]
Thanks both of you for you help. 
Using the missfits package I managed to create new fits file with the header produced by SCAMP. Next problem for me is that I can not use that header to feed to astropy WCS, and I think the problem is that SCAMP is using PV distortion correction and astropy is SIP. I need WCS so I can use for example astropy.allpix2world().
I am sorry for opening issue, this is not an issue, I just did not know where to ask.

**Comment by ebertin** [comment]
Re-opening and moving to discussions.

**Comment by ebertin** [comment]
Hi @tosaric
_afaik_ astropy.wcs [does support the TPV projection](https://docs.astropy.org/en/stable/wcs/supported_projections.html) (since V5.0). Please make sure that SCAMP generates a .head file with `CTYPE1  = "RA---TPV"` and `CTYPE2  = "DEC--TPV"`. For compatibility reasons with older versions (before `TPV` became a standard), SCAMP generates the same CTYPEs  in output as in input. To make sure that the output is TPV, use the option `PROJECTION_TYPE TPV`.
